sdk/src/main/scala/io/horizen/account/state/McAddrOwnershipMsgProcessor.scala

```diff
@@ -38,24 +40,65 @@ case class McAddrOwnershipMsgProcessor(params: NetworkParams) extends NativeSmar
+  override def init(view: BaseAccountStateView, consensusEpochNumber: Int): Unit = {
+    if (!isForkActive(consensusEpochNumber)) {
+      log.warn(s"Can not perform ${getClass.getName} initialization, fork is not active")
+      return
+    }
+    else {
+      if (initDone(view)) {
+        throw new MessageProcessorInitializationException("McAddrOwnership msg processor already initialized")
+      }
+    }
 
-  override def init(view: BaseAccountStateView): Unit = {
-    super.init(view)
-    // set the initial value for the linked list last element (null hash)
+    // this creates the native smart contract address in stateDB
+    // We do not call the parent method because in our case the initialization does not happen at genesis state, and someone might
+    // (on purpose or not) already have created the account, maybe from a deployed solidity smart contract
+    if (!view.accountExists(contractAddress)) {
+      view.addAccount(contractAddress, contractCode)
+      log.debug(s"created Message Processor account $contractAddress")
+    } else {
+      val errorMsg = s"Account $contractAddress already exists!! "
+      log.warn(errorMsg)
+      // TODO maybe we can check if there is any balance at this point and transfer it somewhere
+    }
 
-    // check we do not have this key set to any value yet
+    // set the initial value for the linked list last element (null hash)
+    //-------
+    // check if we have this key set to any value
     val initialTip = view.getAccountStorage(contractAddress, LinkedListTipKey)
 
     // getting a not existing key from state DB using RAW strategy as the api is doing
     // gives 32 bytes filled with 0 (CHUNK strategy gives an empty array instead)
-    if (!initialTip.sameElements(NULL_HEX_STRING_32))
-      throw new MessageProcessorInitializationException("initial tip already set")
+    if (!initialTip.sameElements(NULL_HEX_STRING_32)) {
+
+      // This should not happen, unless someone managed to write to the context of this address.
+      // This should not be possible from a solidity smart contract, but could be from a different native smart contract.
+      // Since the initialization is not done at genesis block, to be on the safe side, we chose not to throw an exception
+      // but just log a warning and then overwrite the value.
+
+      val errorMsg = s"Initial tip already set, overwriting it!! "
+      log.warn(errorMsg)
+    }
 
     view.updateAccountStorage(contractAddress, LinkedListTipKey, LinkedListNullValue)
   }
 
+  override def canProcess(msg: Message, view: BaseAccountStateView, consensusEpochNumber: Int): Boolean = {
+    if (isForkActive(consensusEpochNumber)) {
+      // the gas cost of these calls is not taken into account in this case, we are not tracking gas consumption (and
+      // there is not an account to charge anyway)
+      if (!initDone(view))
+        init(view, consensusEpochNumber)
+      super.canProcess(msg, view, consensusEpochNumber)
```

Comment:
Maybe `super.canProcess(msg, view, consensusEpochNumber) `can be moved at the beginning of canProcess, so it returns faster in case the transaction shouldn't be handled by this message processor.

Reply:
Ok, I reworked a bit this method